generate-offline-map-using-workmanager/src/main/java/com/esri/arcgismaps/sample/generateofflinemapusingworkmanager/WorkerNotification.kt

```diff
@@ -0,0 +1,143 @@
+        // build using the default notification builder with the status message
+        val notification = getDefaultNotificationBuilder(
+            setOngoing = false,
+            contentText = message
+        ).build().apply {
+            // this flag dismisses the notification on opening
+            flags = Notification.FLAG_AUTO_CANCEL
+        }
+
+        with(NotificationManagerCompat.from(applicationContext)) {
+            // cancel the visible progress notification using its id
+            cancel(notificationId)
+            // post the new status notification with a new notificationId
+            notify(notificationId + 1, notification)
+        }
+    }
+
+    /**
+     * Creates a new notification channel and adds it to the NotificationManager
+     */
+    private fun createNotificationChannel() {
+        // get the channel properties from resources
+        val name = applicationContext.getString(R.string.notification_channel_name)
+        val descriptionText =
+            applicationContext.getString(R.string.notification_channel_description)
+        val importance = NotificationManager.IMPORTANCE_HIGH
+        // create a new notification channel with the properties
+        val channel = NotificationChannel(notificationChannelId, name, importance).apply {
+            description = descriptionText
+        }
+        // get the notification system service as a NotificationManager
+        val notificationManager =
+            applicationContext.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        // Add the channel to the NotificationManager
+        notificationManager.createNotificationChannel(channel)
+    }
+
+    /**
+     * Creates and returns a new NotificationCompat.Builder with the given [contentText]
+     * and as an ongoing notification based on [setOngoing]
+     */
+    private fun getDefaultNotificationBuilder(
+        setOngoing: Boolean,
+        contentText: String
+    ): NotificationCompat.Builder {
+        return NotificationCompat.Builder(applicationContext, notificationChannelId)
+            // sets the notifications title
+            .setContentTitle(applicationContext.getString(R.string.notification_title))
+            // sets the content that is displayed on expanding the notification
+            .setContentText(contentText)
+            .setSmallIcon(R.drawable.ic_launcher_foreground)
+            // sets it to only show the notification alert once, in case of progress
+            .setOnlyAlertOnce(true)
+            .setCategory(NotificationCompat.CATEGORY_PROGRESS)
+            // ongoing notifications cannot be dismissed by swiping them away
+            .setOngoing(setOngoing)
+            // sets the onclick action to launch the mainActivityIntent
+            .setContentIntent(mainActivityIntent)
+            // sets it to show the notification immediately
+            .setForegroundServiceBehavior(NotificationCompat.FOREGROUND_SERVICE_IMMEDIATE)
```

Comment:
```suggestion
            .setForegroundServiceBehavior(NotificationCompat.FOREGROUND_SERVICE_IMMEDIATE)
            .addAction(R.drawable.ic_launcher_foreground,"Cancel",cancelIntent)
```
Suggestion, but would be okay without it as well.

Since the sample showcases the job worker and running the offline map job in the background, it would kinda be nice to show the cancellation of the job also handled in the background. There is a helpful way to add "[actions](https://developer.android.com/develop/ui/views/notifications/build-notification#Actions)" to notifications which takes also would use a `PendingIntent`.

Reply:
Thanks for the suggestion. I've added a cancellation action.